### PR TITLE
Chat component features editions and bug fixes

### PR DESCRIPTION
**Pull request by chandu379**
1)Added animations for intital message load, thumbnail image load,full image display in modal window and typing event in chat interface.
2)Added fix for chat window height match with side bar
3)Added fix for navbar height,  which is overlapping over chat window in devices with high viewport height like ipad and other ios devices.(only in chat component)
4)Modified socket typing event so as to display mutiple user's typing, by string concatenation
5)Added code for displaying thumbnails with fixed size along with animating icon on load.
6)Added fix for window scroll bar which goes to top on load of more messages(ideally it should be at the same location and user should scroll to top to view loaded ones)

### CODE REVIEW

File: src/client/app/chat/chat.component.css [hunk 1]
```diff
@@ -672,4 +675,105 @@ mm-prescription {
...
+        -webkit-transform: scale(0);
+        transform: scale(0);
+    }
+    40% {
+        -webkit-transform: scale(1.0);
+        transform: scale(1.0);
+    }
+}
+
+.lds-ring {
```
Comment (arun-awnics) [inline src/client/app/chat/chat.component.css]: What does lds stand for?

File: src/client/app/chat/chat.component.ts [hunk 2]
```diff
@@ -108,6 +109,7 @@ export class ChatComponent implements OnInit {
   doctorDetails:any;
   showPrescriptionComponent:Boolean = false;
   digitalSignature:string;
+  displayMessageLoadingIcon:Boolean ;
```
Comment (arun-awnics) [inline src/client/app/chat/chat.component.ts]: rename it to displayLoader or simply loader

Comment (chandu379) [reply src/client/app/chat/chat.component.ts]: But there are multiple animation loaders in chat component

File: src/client/app/chat/chat.component.css (hunk 1 above)
Comment (chandu379) [reply src/client/app/chat/chat.component.css]: may be loadspinner. i haven't designed that

File: src/client/app/shared/message-components/image-message.component.ts [hunk 3]
```diff
@@ -19,27 +19,42 @@ export class ImageMessageComponent implements OnInit {
...
 
-    downloadImage(fileName: string) {
-        this.chatService.downloadFile(fileName)
+    downloadThumbImage(imageName: string) {
+        this.imageName = imageName;
+        this.chatService.downloadFile(imageName)
             .subscribe((res) => {
                 res.onloadend = () => {
-                    this.url = this.sanitizer.bypassSecurityTrustUrl(res.result);
+                    this.thumbImageUrl = this.sanitizer.bypassSecurityTrustUrl(res.result);
```
Comment (arun-awnics) [inline src/client/app/shared/message-components/image-message.component.ts]: You may use safe pipe to bypassSecurity

Comment (chandu379) [reply src/client/app/shared/message-components/image-message.component.ts]: That is the same old code i just renamed those variables to  convey better.I think bypassSecurityTrustUrl also produces a safe  url

**Pull request closed without merging**